common/src/main/java/org/apache/cassandra/sidecar/common/NodeSettings.java

```diff
@@ -16,33 +16,75 @@
  * limitations under the License.
  */
 
-
 package org.apache.cassandra.sidecar.common;
 
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Holds information about the specific node settings
  */
 public class NodeSettings
 {
-    private final String releaseVersion;
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeSettings.class);
+    private static final String SIDECAR_VERSION = getSidecarVersion();
+
     private final String partitioner;
+    private final String releaseVersion;
+    private final String sidecarVersion;
+
+    private static String getSidecarVersion()
+    {
+        try (InputStream version = NodeSettings.class.getResourceAsStream("/sidecar.version"))
+        {
+            return IOUtils.toString(version, StandardCharsets.UTF_8);
+        }
+        catch (Exception exception)
+        {
+            LOGGER.error("Failed to ex Sidecar version", exception);
+        }
+        return "unknown";
+    }
 
     /**
-     * Constructs a new {@link NodeSettings} object with the Cassandra node's release version and partitioner
-     * information.
+     * Constructs a new {@link NodeSettings} object with the Cassandra node's partitioner
+     * and release version information, uses Sidecar version from the currently loaded binary
      *
+     * @param partitioner    the partitioner used by the Cassandra node
      * @param releaseVersion the release version of the Cassandra node
+     */
+    public NodeSettings(String partitioner, String releaseVersion)
+    {
+        this(partitioner, releaseVersion, SIDECAR_VERSION);
+    }
+
+    /**
+     * Constructs a new {@link NodeSettings} object with the Cassandra node's partitioner,
+     * release version, and Sidecar version information
+     *
      * @param partitioner    the partitioner used by the Cassandra node
+     * @param releaseVersion the release version of the Cassandra node
+     * @param sidecarVersion the version of the Sidecar on the Cassandra node
      */
-    public NodeSettings(@JsonProperty("releaseVersion") String releaseVersion,
-                        @JsonProperty("partitioner") String partitioner)
+    public NodeSettings(@JsonProperty("partitioner")    String partitioner,
+                        @JsonProperty("releaseVersion") String releaseVersion,
```

Comment:
What is the reason of switching the order? It causes other refactorings in the same PR.

Reply:
I wanted to keep two version numbers (of Cassandra and of Sidecar) next to each other in method signatures.

Reply:
I do not think the order is fixed even if you change the order in the param list, unless `@JsonPropertyOrder` is declared. 

It kind of reminds me of the clarity of the payload. See the example below. It feels unclear. Because the fields in node settings response payload were only about the cassandra node below. Now, we have a field about sidecar, that is sitting next to the cassandra fields at the _same_ level. 
We certainly cannot change the schema of the existing fields. I guess you do not want to add an extra endpoint just for sidecar. 
What do you think about adding a sidecar json object, that includes the sidecar version and other related settings to be added in the future? 

```json
{
  "partitioner": "Murmur3Partitioner",
  "releaseVersion": "4.0",
  "sidecarVersion": "1.0"
}
```

The alternative,
```java
{
  "partitioner": "Murmur3Partitioner",
  "releaseVersion": "4.0",
  "sidecar": {
    "version": "1.0"
  }
}
```

Reply:
Honestly, this feels like a bit of an overkill to me, as there's a chance there will be no other Sidecar-related values…
But I can introduce another class just for this one attribute, if you insist.

That being said, I don't think changing the schema of the existing is completely out of the question.
This code was published very recently and is probably not actively used by anyone, so we could try renaming `releaseVersion` into `cassandraVersion`…

Reply:
Changed the Sidecar version field type as per our conversation, let me know whether this is what you had in mind.